src/Simplify_Select.cpp

```diff
@@ -152,6 +184,9 @@ Expr Simplify::visit(const Select *op, ExprInfo *bounds) {
              rewrite(select(x, y + z, y), y + select(x, z, 0)) ||
              rewrite(select(x, y, y + z), y + select(x, 0, z)) ||
 
+             rewrite(select(x, y - z, y), y + select(x, -z, 0), y != 0) ||
```

Comment:
I think you want !is_const(y) as the predicate for this and the next rule.

Reply:
That wouldn't work for select(x, 3 - z, 3) though, which is I think what I actually added this rule for?

Reply:
Discussed offline, summarized here: that would increase op count, and it would be perfectly legal for another rule to fold the constant right back in, forming a cycle.